lms/djangoapps/courseware/testutils.py

```diff
@@ -90,27 +118,60 @@ def setup_user(self, admin=False, enroll=False, login=False):
         if login:
             self.login()
 
+    def set_block_to_be_requested(self, block_name):
+        """
+        Sets the block currently being tested.  This allows the positive and
+        negative tests to refer to the same block.
+
+        Arguments:
+            block_name: The name of the block (e.g. 'html_block'), that should
+                match both the attribute for the block as well as the key into
+                BLOCK_SPECIFIC_ELEMENTS.  Use None to clear values.
+
+        Side Effects:
+            Sets block_to_be_requested to the block under test.
+            Sets block_specific_chrome_html_elements to an array of elements that
+               should appear in the courseware chrome specifically for this block.
+        """
+        if block_name is None:
+            self.block_to_be_requested = None  # pylint: disable=attribute-defined-outside-init
+            self.block_specific_chrome_html_elements = None  # pylint: disable=attribute-defined-outside-init
+        else:
+            self.block_to_be_requested = getattr(self, block_name)  # pylint: disable=attribute-defined-outside-init
+            block_specific_elements = self.BLOCK_SPECIFIC_ELEMENTS[block_name]['courseware_chrome_html_elements']
+            self.block_specific_chrome_html_elements = block_specific_elements  # pylint: disable=attribute-defined-outside-init
```

Comment:
Nit: Either disable this pylint violation at the top of the file or at the top of this method - so it's not as noisy in the code.  `attribute-defined-outside-init` is a typical pylint violation that tests are exempted from following.